syntax = "proto2";

package model.detection.protos;

import "model/detection/protos/eval.proto";
import "model/detection/protos/input_reader.proto";
import "model/detection/protos/model.proto";
import "model/detection/protos/train.proto";

// Convenience message for configuring a training and eval pipeline. Allows all
// of the pipeline parameters to be configured from one file.
message TrainEvalPipelineConfig {
    optional DetectionModel model = 1;
    optional TrainConfig train_config = 2;
    optional InputReader train_input_reader = 3;
    optional EvalConfig eval_config = 4;
    optional InputReader eval_input_reader = 5;
}
